support-frontend/assets/components/subscriptionCheckouts/address/addressFields.tsx

```diff
@@ -184,10 +189,47 @@ export function AddressFields({ scope, ...props }: PropTypes): JSX.Element {
 				data-qm-masking="blocklist"
 				label="Country"
 				value={props.country}
-				onChange={(e) => {
-					const isoCountry = Country.fromString(e.target.value);
-					if (isoCountry) {
-						props.setCountry(isoCountry);
+				onChange={(event) => {
+					const selectedCountry = Country.fromString(event.target.value);
+
+					if (selectedCountry && countryGroupId) {
+						const selectedCountryGroup = Object.entries(countryGroups).find(
+							([, countryGroup]) =>
+								countryGroup.countries.includes(selectedCountry),
+						)?.[0];
+
+						if (countryGroupId !== selectedCountryGroup) {
+							const pathname = window.location.pathname;
+							const currentInternationalisationId = pathname.split('/')[1];
+							const selectedInternationalisationId =
+								countryGroups[selectedCountryGroup as CountryGroupId]
+									.supportInternationalisationId;
+							const redirectPathname = pathname.replace(
+								currentInternationalisationId,
+								selectedInternationalisationId,
+							);
+
+							const urlSearchParams = new URLSearchParams(
+								window.location.search,
+							);
+							const product = urlSearchParams.get('product') as ProductKey;
+							const ratePlan = urlSearchParams.get('ratePlan') as string;
+							const { productKey, ratePlanKey } =
+								internationaliseProductAndRatePlan(
+									selectedInternationalisationId,
+									product,
+									ratePlan,
+								);
+							urlSearchParams.set('product', productKey);
+							urlSearchParams.set('ratePlan', ratePlanKey);
+
+							const location = `${redirectPathname}?${urlSearchParams.toString()}`;
+							window.location.href = location;
+						}
+					}
+
+					if (selectedCountry) {
```

Comment:
I think the page reload overrides this selection. ie selecting `Taiwan` from within Uk region causes path to be updated to `int` and the resulting reload replaces Taiwan with `India`.

Reply:
I've added the `country` URL parameter and that's fixed it.

Reply:
Selecting Taiwan will redirect to the `/int` country group checkout, the default country for this is India, if you're checking out out locally (eg. in the UK) on the `/int` checkout it'll set your delivery/billing country to India - if you're in Taiwan and you're on the `/int` checkout it should identify via your geo location that you're in Taiwan the  delivery/billing country will be set to Taiwan.

Reply:
> Selecting Taiwan will redirect to the `/int` country group checkout, the default country for this is India, if you're checking out out locally (eg. in the UK) on the `/int` checkout it'll set your delivery/billing country to India - if you're in Taiwan and you're on the `/int` checkout it should identify via your geo location that you're in Taiwan the delivery/billing country will be set to Taiwan.

Saying that if you're geo location is Taiwan you should've ended up on the `/int` checkout anyway, forcing the country with the `country` query string param as @jamesgorrie has done seems like a safe way of ensuring the country is correctly set in the reloaded form,